library/src/main/java/com/novoda/downloadmanager/LiteFileDownloadError.java

```diff
@@ -0,0 +1,66 @@
+package com.novoda.downloadmanager;
+
+class LiteFileDownloadError implements FileDownloader.FileDownloadError {
+
+    private final String rawRequest;
+    private final String message;
+    private final int errorCode;
+
+    LiteFileDownloadError(String rawRequest, String message, int errorCode) {
```

Comment:
what is "Lite" about this? Is it saying its a "Lite File Download" that has Error'd? Or is it saying a File Download has a "Lite Error" which I'm not sure what that is either?

Reply:
`Lite` is used to represent all of our internal classes. We always introduce an interface on this project which represents the publicly available information and then hide everything else. Clients do not have the opportunity to mess with these classes, they only deal with interfaces.

Reply:
hmm strange name.  Two things then:

Shouldn't the interface be declared public:

```
    interface FileDownloadError {
```

and

The `FileDownloadError` javadoc is referencing the internal class, but clients shouldn't be aware of it?
```
    /**
     * Represents the information of a {@link LiteFileDownloadError} that is accessible to clients.
     */
    interface FileDownloadError {
```

Reply:
Good point about referencing the internal class 👍 

The interface, `FileDownloadError` is public, it's enclosed by another interface which is public.

Reply:
6d130c92186cf7b0ac9f52a629ed4d84c2de7cd6